numpy/testing/utils.py

```diff
@@ -930,7 +931,7 @@ def compare(x, y):
             if npany(gisinf(x)) or npany( gisinf(y)):
                 xinfid = gisinf(x)
                 yinfid = gisinf(y)
-                if not xinfid == yinfid:
+                if not (xinfid == yinfid).all():
```

Comment:
Out of curiousity, why did this not create an error before? It seems obviously wrong?